darwin/importer/formats/superannotate.py

```diff
@@ -0,0 +1,159 @@
+import json
+from functools import partial
+from os.path import join
+from pathlib import Path
+from typing import Any, Callable, Dict, Iterable, List, Optional, Set, cast
+
+from jsonschema import validate
+
+from darwin.datatypes import Annotation, AnnotationClass, AnnotationFile, make_keypoint
+from darwin.importer.formats.superannotate_schemas import (
+    classes_export,
+    superannotate_export,
+)
+
+
+def parse_path(path: Path) -> Optional[AnnotationFile]:
+    """
+    Parses SuperAnnotate annotations inside the given file and returns the corresponding darwin 
+    annotations. If the given file is not a ``.json`` or is a ``classes.json`` then ``None`` is 
+    returned instead. 
+
+    Each annotation file must have a structure simillar to the following:
+    
+    .. code-block:: javascript
+        {
+            "instances": [
+                {
+                    "type": "point",
+                    "x": 1,
+                    "y": 0
+                },
+                // { ... }
+            ],
+            "metadata": {
+                "name": "a_file_name.json"
+            }
+        }
+    
+    Currently we support the following annotations:
+
+        - point ``Vector``: https://doc.superannotate.com/docs/vector-json#point
+    
+
+    Each file must also have in the same folder a ``classes.json`` file with information about 
+    the classes. This file must have a structure simillar to:
+
+    .. code-block:: javascript
+        [
+            {"name": "a_name_here", "id": 1},
+            // { ... }
+        ]
+
+    You can check the SuperAnnotate Schemas in ``superannotate_schemas.py``.
+
+    Parameters
+    --------
+    path: Path
+        The path of the file to parse.
+    
+    Returns
+    -------
+    Optional[darwin.datatypes.AnnotationFile]
+        The AnnotationFile with the parsed information from each SuperAnnotate annotation inside 
+        or ``None`` if the given file is not a ``.json`` or is ``classes.json``.
+    
+    Raises
+    ------
+    ValidationError
+        If any given JSON file is malformed or if it has an unknown annotation. 
+        To see a list of possible annotation formats go to:
+        https://doc.superannotate.com/docs/vector-json
+    """
+
+    if not _is_annotation(path):
+        return None
+
+    classes_path = Path(join(path.parent, "classes.json"))
+    if not classes_path.is_file():
+        raise ValueError("Folder must contain a 'classes.json' file with classes information.")
+
+    with classes_path.open() as classes_file:
+        classes = json.load(classes_file)
+        validate(classes, schema=classes_export)
+
+        with path.open() as annotation_file:
+            data = json.load(annotation_file)
+            validate(data, schema=superannotate_export)
+
+            instances: List[Dict[str, Any]] = data.get("instances")
+            metadata: Dict[str, Any] = data.get("metadata")
+
+            return _convert(instances, path, classes, metadata)
```

Comment:
I guess we don't need to keep this block at this level of indentation, we can go one level back

Reply:
How can we go back 1 level? Can I open 2 files at the same time with `with` ?

Reply:
done!